### ISSUE TITLE
Show more descriptive errors when asking for events but there is no connection

### ISSUE THREAD

**Issue by Euen** [inaka/shotgun#95]
When you run

```erlang
Headers = #{ <<"content-type">> => <<"application/json">>
           , <<"Cookie">> => list_to_binary(CookieString)},

{ok, Pid} = shotgun:open("localhost", 8080),

{ok, #{status_code := 401}} =
      shotgun:get( Pid
                 , "/events"
                 , Headers
                 , #{ async => true
                    , async_mode => sse}),

shotgun:events(Pid),
```
And have no sse connection (perhaps because of a 401 error) you get this strange error

```erlang
** exception exit: {{badarg,[{queue,to_list,
                                    [{#Ref<0.0.1.149676>,
                                      {ok,#{headers => [{<<"connection">>,<<"keep-alive">>},
                                             {<<"server">>,<<"Cowboy">>},
                                             {<<"date">>,<<"Fri, 24 Jul 2015 14:08:24 GMT">>},
                                             {<<"content-length">>,<<"0">>}],
                                            status_code => 401}}}],
                                    [{file,"queue.erl"},{line,87}]},
                             {shotgun,handle_sync_event,4,
                                      [{file,"/Users/euen/Documents/fiar/deps/shotgun/src/shotgun.erl"},
                                       {line,289}]},
                             {gen_fsm,handle_msg,7,[{file,"gen_fsm.erl"},{line,503}]},
                             {proc_lib,init_p_do_apply,3,
                                       [{file,"proc_lib.erl"},{line,237}]}]},
                    {gen_fsm,sync_send_all_state_event,[<0.4453.0>,get_events]}}
```

**Comment by kennethlakin** [comment]
Can you re-try this with the latest master branch? I can't reproduce this, but it's *very* likely that I'm failing to properly configure my little HTTP server.

**Comment by kennethlakin** [comment]
Yeah, it looks like https://github.com/inaka/shotgun/blob/a2e90087175d766ee6c09b78949795b48f28dfbf/src/shotgun.erl#L443 is the culprit here. the ```responses``` key is only ever supposed to contain a value of type ```queue:queue()```, but line 443 assigns whatever random stuff ```gen_fsm:reply/2``` (whose return value you're supposed to ignore [0]) hands back to it. At this moment, i'm not sure what the correct thing to do is.

@jfacorro , do you have any insight as to the intent of the code here? I'm going to break for dinner, but will be back in a few hours to keep poking at this, myself.

Also, can Dialyzer understand type specifications for particular map keys? If it can, that might help avoid this kind of mistake in the future.

[0] <http://www.erlang.org/doc/man/gen_fsm.html#reply-2>

**Comment by kennethlakin** [comment]
The troublesome code used to look like this: 
https://github.com/inaka/shotgun/commit/c59cced313e2c6e5a4005e048c66da0f113afb2c#diff-061a9243c5d9756e7d8111fd8f4b4ba7R341 (notice the true branch ends in ```queue:in(Response, Responses)``` )
but was changed to this: https://github.com/inaka/shotgun/commit/0243a27304faf0c58eff74103757eaa2229a37b7#diff-061a9243c5d9756e7d8111fd8f4b4ba7 . The log message leads me to believe that this was an oversight.

PR #115 *seems* to address this issue, but *please* note that I haven't performed extensive testing with any servers that *actually* provide SSEs.

**Comment by jfacorro** [comment]
Fixed by #115
